aptos-move/framework/aptos-framework/sources/datastructures/big_ordered_map.move

```diff
@@ -695,18 +695,78 @@ module aptos_std::big_ordered_map {
 
     /// Apply the function to a reference of each element in the vector.
     public inline fun for_each_ref<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>, f: |&K, &V|) {
-        let iter = self.internal_new_begin_iter();
-        while (!iter.iter_is_end(self)) {
-            f(iter.iter_borrow_key(), iter.iter_borrow(self));
-            iter = iter.iter_next(self);
-        };
+        // let iter = self.internal_new_begin_iter();
+        // while (!iter.iter_is_end(self)) {
+        //     f(iter.iter_borrow_key(), iter.iter_borrow(self));
+        //     iter = iter.iter_next(self);
+        // };
 
         // TODO use this more efficient implementation when function values are enabled.
-        // self.for_each_leaf_node_ref(|node| {
-        //     node.children.for_each_ref(|k: &K, v: &Child<V>| {
-        //         f(k, &v.value);
-        //     });
-        // })
+        self.for_each_leaf_node_children_ref(|children| {
+            children.for_each_ref(|k: &K, v: &Child<V>| {
+                f(k, v.internal_leaf_borrow_value());
+            });
+        })
+    }
+
+    public inline fun zip_for_each_ref<K: drop + copy + store, V1: store, V2: store>(self: &BigOrderedMap<K, V1>, other: &BigOrderedMap<K, V2>, f: |&K, &V1, &V2|) {
```

Comment:
nit: add documentation about the behaviour when maps that are not equal in size?